teamsyncd/teamsync.cpp

```diff
@@ -194,6 +215,41 @@ void TeamSync::removeLag(const string &lagName)
     m_selectablesToRemove.insert(lagName);
 }
 
+#define MAX_PID_STRLEN 10
+void TeamSync::cleanTeamProcesses(int signo)
+{
+    ssize_t sz = 0;
+    int fd, pid;
+    char pid_str[MAX_PID_STRLEN] = {0};
+    char *endptr = NULL;
+    string file_path;
+
+    for (auto it=m_teamSelectables.begin(); it!=m_teamSelectables.end(); ++it)
+    {
+       /* There are PID files created for each teamd process in /var/run/teamd DIR */
+       file_path = string(PID_FILE_PATH) + it->first + string(".pid");
+
+       /* Open the file, get the PID and send signal to the process */
+       fd = open(file_path.c_str(), O_RDONLY);
+       if (fd > 0)
+       {
```

Comment:
should we report if fd < 0?

Reply:
Same as above comment to minimize logging as this function was added as a Signal handler. I will add a Log message in the TeamPortSync destructor, will be helpful in debugging -- logging the cleanup of each Portchannel.

Reply:
Added more Logs now. Please take a look